scripts/update.py

```diff
@@ -0,0 +1,454 @@
+    cores = os.cpu_count()
+    print("threads: ", cores)
+
+    pool = Pool(processes=cores)
+    progress = tqdm.tqdm(work_list, miniters=len(work_list) / 200)
+    fail = False
+    stats_acc = dict()
+    stats_nondeter_keys = {
+        "dse.NumDomMemDefChecks", "ir.NumInstrRenumberings",
+        "basicaa.SearchTimes", "aa.NumMayAnlias",
+        "capture-tracking.NumCaptured", "aa.NumMustAlias",
+        "memory-builtins.ObjectVisitorArgument", "aa.NumNoAlias",
+        "assume-queries.NumAssumeQueries", "capture-tracking.NumNotCaptured",
+        "ipt.NumInstScanned"
+    }
+
+    comptime_res = []
+    with open("test.log", "w") as log:
+        for fn, status, res, stats in pool.imap_unordered(run_opt, work_list):
+            fn = os.path.relpath(fn, BENCH_DIR)
+            fn = fn.replace("/original/", "/")
+            if status != "success":
+                progress.write(fn + " " + status)
+                log.write(fn + " " + status + "\n")
+                fail = True
+            else:
+                if COMPTIME_OUT:
+                    comptime_res.append((fn, res))
+                if STATS_OUT:
+                    for k in stats:
+                        if k in stats_nondeter_keys:
+                            continue
+                        stats_acc[k] = stats_acc.get(k, 0) + stats[k]
+            progress.update()
+        progress.close()
+
+    if COMPTIME_OUT is not None:
+        comptime_res.sort(key=lambda x: x[0])
+        with open(COMPTIME_OUT, "w") as f:
+            for k, v in comptime_res:
+                f.write(f"{k} {v}\n")
+    if STATS_OUT is not None:
+        with open(STATS_OUT, "w") as f:
+            json.dump(stats_acc, f, indent=2, sort_keys=True)
+
+    return not fail
+
+
+def os_do(cmd):
+    okay = os.system(cmd) & 0xff == 0
+    if not okay:
+        print("Failed: '{}'".format(cmd))
+    return okay
+
+
+def update_baseline():
+    global OUT
+    global LLVM_REV
+    if COMPTIME_OUT is not None:
+        if not finalize_baseline_comptime():
+            return False
+    if STATS_OUT is not None:
+        if not finalize_baseline_stats():
+            return False
+
+    if not OUT:
+        return True
+
+    if not os_do("git add ."):
+        return False
+
+    if not os_do(
+            "git commit -m 'llvm: Update baseline to {}'".format(LLVM_REV)):
+        return False
+
+    if not os_do("git push"):
+        return False
+
+    return True
+
+
+def update_pr():
+    global OUT
+    global RUNNER_ID
+    global LLVM_REV
+
+    diff_stat = run_cmd("git diff --shortstat", 1800.0)
+    if not os_do("./scripts/filter_pr_changes.py"):
+        return False
+    if not os_do("git commit -m 'pre-commit: Update'"):
+        return False
+
+    if not os_do("git push"):
+        return False
+
+    lines = []
+    lines.append("runner: {}".format(RUNNER_ID))
+    lines.append(
+        "baseline: https://github.com/llvm/llvm-project/commit/{}".format(
+            LLVM_REV))
```

Comment:
```suggestion
    lines.append(
        f"baseline: https://github.com/llvm/llvm-project/commit/{LLVM_REV}")
```
Use f-string.

Reply:
Only here? Does it really matter (I tend to dislike f-string, it breaks some of my tooling: https://github.com/google/yapf/issues/1250)

Reply:
up to you

Reply:
Pref is leave as is but I don't really care